File: src/main/java/com/basil/shoppingcart/service/ProductService.java
package com.basil.shoppingcart.service;

import java.math.BigDecimal;
import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import com.basil.shoppingcart.dto.request.ProductRequest;
import com.basil.shoppingcart.dto.response.ProductResponse;

public interface ProductService {

    ProductResponse createProduct(ProductRequest request);

    ProductResponse getProductById(Long id);

    List<ProductResponse> getAllProducts();

    Page<ProductResponse> getProducts(
            String name,
            String category,
            String brand,
            BigDecimal minPrice,
            BigDecimal maxPrice,
            Boolean inStock,
            Pageable pageable
    );

    ProductResponse updateProduct(
            Long id,
            ProductRequest request
    );

    void deleteProduct(Long id);
}

File: shopping-cart-frontend/src/services/productService.js
import apiClient from "./apiClient";


const productService = {


    getProducts: async (params = {}) => {

        const response =
            await apiClient.get(
                "/products/search",
                {
                    params
                }
            );

        return response.data;
    },


    getAllProducts: async () => {

        const response =
            await apiClient.get(
                "/products"
            );

        return response.data;
    },


    getProductById: async (id) => {

        const response =
            await apiClient.get(
                `/products/${id}`
            );

        return response.data;
    },

    deleteProduct: async (id) => {
        const response = await apiClient.delete(`/products/${id}`);
        return response.data;
    },

    updateProduct: async (id, productData) => {
        const response = await apiClient.put(`/products/${id}`, productData);
        return response.data;
    },

    createProduct: async (productData) => {
        const response = await apiClient.post("/products", productData);
        return response.data;
    }

};


export default productService;